---
layout: page
title: R 
---

### Developed and maintained:

   - [ZIQ-SIR](https://github.com/tianyingw/ZIQ-SIR) _(ZIQ-SIR)_: We provide the R code for a semiparametric quantile rank score test for zero-inflated data.

   - [ZIQSI](https://github.com/tianyingw/ZIQSI) _(ZIQSI)_: This R package is designed to provide advanced statistical tools for quantile curve fitting and average quantile effect calculation via a quantile single index model.

   - [MGQDA](https://github.com/tianyingw/MGQDA/tree/main) _(MGQDA)_: An R package for multi-group high-dimensional discriminant analysis.

   - [HighCal](http://github.com/tianyingw/HighCal) _(HighCal)_: An R package that implements the high-dimensional calibration method for contaminated data.
     
   - [ZIKQ](https://github.com/tianyingw/ZIKQ) _(ZIKQ)_: We provide the R code for constructing centile charts for natural history data. It implements the method from [ZIKQ: An innovative centile chart method for utilizing natural history data in rare disease clinical development]().
   
   - [SIMFEX](https://github.com/tianyingw/SIMFEX) _(SIMFEX)_: We provide a SIMulation-Free EXtrapolation method for misspecified models with errors-in-variables.
   
   - [ZIPG](https://cran.r-project.org/web/packages/ZIPG/index.html) _(Zero-Inflated Poisson-Gamma)_: The R package ZIPG provides a flexible Zero-inflated Poisson-Gamma Model model for microbiome count data by connecting both mean abundance and the variability to different covariates. It implements the method from [A Flexible Zero-Inflated Poisson-Gamma Model with Application to Microbiome Read Count Data (2022)](https://www.tandfonline.com/doi/abs/10.1080/01621459.2022.2151447?journalCode=uasa20).
   
   - [QTWAS](https://github.com/tianyingw/QTWAS) _(Quantile Transcriptome-Wide Association Analysis)_: We provide a quantile tool for investigating nonlinear gene-trait associations in Transcriptome-Wide Association Analysis (TWAS). It implements the method from [A unified quantile framework reveals nonlinear heterogeneous transcriptome-wide associations (2022+)](https://arxiv.org/pdf/2207.12081.pdf). More information can be found on [this webpage](https://tianyingw.github.io/QTWAS/).
  
   - [iQRAT](https://github.com/tianyingw/iQRAT) _(Integrated Quantile RAnk Test)_: The R package iQRAT provides an efficient quantile rank test for heterogeneous association in sequencing study. It implements the method from [Integrated Quantile RAnk Test (iQRAT) for the heterogeneous joint effect of rare and common variants in sequencing studies (2022)](https://arxiv.org/abs/1910.10102).

   - [IQKM](https://github.com/tianyingw/IQKM) _(Integrated Quantile Kernel Machine)_: The R package IQKM provides tools for testing microbiome-outcome association. It implements the method from [Testing microbiome association using integrated quantile regression models (2022)](https://academic.oup.com/bioinformatics/advance-article-abstract/doi/10.1093/bioinformatics/btab668/6374494).

   - [DAP](https://cran.r-project.org/web/packages/DAP/index.html) _(Discriminant Analysis via Projection)_: The R package DAP provides tools for high-dimensional binary classification in the case of unequal covariance matrices. It implements the method from the following paper:
[Sparse quadratic classification rules via linear dimension reduction by Gaynanova and Wang (2019)](https://arxiv.org/abs/1711.04817).

  - [CCP](https://github.com/tianyingw/CCP) _(Categorize a Continuous Predictor)_: The R package CCP provides tools for correcting the bias due to measurement error when the continuous risk predictor is categorized. The package includes logistic and linear regression, considering when external data are or are not provided. It implements the method from [categorizing a continuous predictor subject to measurement error (2018)](https://projecteuclid.org/euclid.ejs/1544518836).
    

### Contributed:

  - [deconvolve](https://rdrr.io/github/TimothyHyndman/deconvolve/man/deconvolve-package.html): The R package deconvolve provides tools for performing non-parametric deconvolution on measurement error problems. It contains functions for finding bandwidths, deconvolved densities and non-parametric regression estimates.

  - [MiRKAT](https://cran.r-project.org/web/packages/MiRKAT/index.html): Test for overall association between microbiome composition data and phenotypes via phylogenetic kernels. It includes the method from [Testing microbiome association using integrated quantile regression models (2022)](https://academic.oup.com/bioinformatics/advance-article-abstract/doi/10.1093/bioinformatics/btab668/6374494).
  
  - [Optimal pre-post allocation](https://doi.org/10.5281/zenodo.7594938): R codes for reproducing numerical studies in the paper [The optimal pre-post allocation for randomized clinical trials (2023)](https://bmcmedresmethodol.biomedcentral.com/articles/10.1186/s12874-023-01893-w).

