---
layout: page
title: Software
---

### R packages:

- _Developed and maintained:_
   
   - [QTWAS](https://github.com/tianyingw/QTWAS) _(Quantile Transcriptome-Wide Association Analysis)_: We provide a quantile tool for investigating nonlinear gene-trait associations in Transcriptome-Wide Association Analysis (TWAS). It impelents the method from [A unified quantile framework reveals nonlinear heterogeneous transcriptome-wide associations (2022+)](https://arxiv.org/pdf/2207.12081.pdf). More information can be found on [this webpage](https://tianyingw.github.io/QTWAS/).
  
   - [iQRAT](https://github.com/tianyingw/iQRAT) _(Integrated Quantile RAnk Test)_: The R package iQRAT provides an efficient quantile rank test for heterogeneous association in sequencing study. It implements the method from [Integrated Quantile RAnk Test (iQRAT) for heterogeneous joint effect of rare and common variants in sequencing studies (2022)](https://arxiv.org/abs/1910.10102).

   - [IQKM](https://github.com/tianyingw/IQKM) _(Integrated Quantile Kernel Machine)_: The R package IQKM provides tools for testing microbiome-outcome association. It impelements the method from [Testing microbiome association using integrated quantile regression models (2021)](https://academic.oup.com/bioinformatics/advance-article-abstract/doi/10.1093/bioinformatics/btab668/6374494).

   - [DAP](https://cran.r-project.org/web/packages/DAP/index.html) _(Discriminant Analysis via Projection)_: The R package DAP provides tools for high-dimensional binary classification in the case of unequal covariance matrices. It implements the method from the following paper:
[Sparse quadratic classification rules via linear dimension reduction by Gaynanova and Wang (2019)](https://arxiv.org/abs/1711.04817).

  - [CCP](https://github.com/tianyingw/CCP) _(Categorize a Continuous Predictor)_: The R package CCP provides tools for correcting the bias due to measurement error when the continuous risk predictor is categorized. The package includes logistic and linear regression, considering when external data are or are not provided. It implements the method from [categorizing a continuous predictor subject to measurement error (2018)](https://projecteuclid.org/euclid.ejs/1544518836).
    

- _Contributed:_

  - [deconvolve](https://rdrr.io/github/TimothyHyndman/deconvolve/man/deconvolve-package.html): The R package deconvolve provides tools for performing non-parametric deconvolution on measurement error problems. It contains functions for finding bandwidths, deconvolved densities and non-parametric regression estimates.

  - [MiRKAT](https://cran.r-project.org/web/packages/MiRKAT/index.html): Test for overall association between microbiome composition data and phenotypes via phylogenetic kernels. It includes the method from [Testing microbiome association using integrated quantile regression models (2021)](https://academic.oup.com/bioinformatics/advance-article-abstract/doi/10.1093/bioinformatics/btab668/6374494).

### Shiny app:

- [QTWAS Genome Browser](https://tianyingw.shinyapps.io/QTWAS/): Visualize quantile TWAS results in [A unified quantile framework reveals nonlinear heterogeneous transcriptome-wide associations (2022+)](https://arxiv.org/pdf/2207.12081.pdf).
